src/sonic-ctrmgrd/ctrmgr/container_state

```diff
@@ -0,0 +1,317 @@
+
+    state_db = ConfigDBConnector()
+    state_db.db_connect("STATE_DB", wait_for_init=False, retry_on=True)
+    tbl = state_db.get_table('FEATURE')
+    if feature in tbl:
+        state_data.update(tbl[feature])
+
+
+def read_field(feature, field, default):
+    tbl = state_db.get_table('FEATURE')
+    if (feature in tbl) and (field in tbl[feature]):
+        return tbl[feature][field]
+    else:
+        return default
+
+def check_unset_labels(feature, version):
+    tbl = state_db.get_table(KUBE_LABEL_TABLE)
+    if KUBE_LABEL_UNSET_KEY in tbl:
+        labels = tbl[KUBE_LABEL_UNSET_KEY]
+    else:
+        labels = {}
+    return _get_version_key(feature, version) in labels
+
+
+def drop_label(feature, version):
+    """ Set/drop label as required
+        Update is done in state-db.
+        ctrmgrd sets it with kube API server as required
+    """
+    name = _get_version_key(feature, version)
+    state_db.mod_entry(KUBE_LABEL_TABLE, KUBE_LABEL_UNSET_KEY, { name: "" })
+
+
+def update_data(feature, data):
+    debug_msg("{}: {}".format(feature, str(data)))
+    state_db.mod_entry("FEATURE", feature, data) 
+
+    
+def get_docker_id():
+    cmd = 'cat /proc/self/cgroup | grep -e ":memory:" | rev | cut -f1 -d\'/\' | rev'
+    proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+    output = proc.communicate()[0].decode("utf-8")
+    return output.strip()[:12]
+
+
+def to_int(s):
+    try:
+        return(int(s))
+    except ValueError as ex:
+        syslog.syslog(syslog.LOG_ERR, "Non convertible err={}".format(str(ex)))
+        return 0
+
+
+def instance_lower(feature, version):
+    if ((state_data[REMOTE_STATE] == "none") or
+            (state_data[REMOTE_STATE] == "stopped")):
+        # No one running to compare version
+        return False
+
+    ct_version = state_data[VERSION]
+    ct = ct_version.split('.') if ct_version else "0.0.0".split('.')
+    nxt = version.split('.') if version else "0.0.0".split('.')
+    ret = False
+    for cs, ns in zip(ct, nxt):
+        c = to_int(cs)
+        n = to_int(ns)
+        if n < c:
+            ret = True
+            break
+        elif n > c:
+            break
+
+    debug_msg("compare version: new:{} current:{} res={}".format(
+        version, ct_version, ret))
+    return ret
+
+
+def is_active(feature):
+    if state_data[SYSTEM_STATE] == "up":
+        return True
+    else:
+        syslog.syslog(syslog.LOG_ERR, "Found inactive for {}".format(feature))
+        return False
+
+
+def update_state(args, is_up):
+    data = {
+            CURRENT_OWNER: args.owner if is_up else "none",
+            DOCKER_ID: get_docker_id() if is_up else "",
+            UPD_TIMESTAMP: str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+            }
+    if is_up:
+        data[VERSION] = args.version
+
+        if (args.owner == "local"):
+            # Disable deployment of this version as available locally
+            drop_label(args.feature, args.version)
```

Comment:
What is the difference between drop_label here and set_label(feature, False) in container_action start?